backend/capellacollab/core/authentication/database/__init__.py

```diff
@@ -19,135 +20,140 @@
 from capellacollab.users.models import Role
 
 
-def verify_admin(token=Depends(JWTBearer()), db=Depends(get_db)):
-    if not is_admin(token, db):
-        raise HTTPException(
-            status_code=403,
-            detail={
-                "reason": "The role administrator is required for this transaction.",
-            },
+class RoleVerification:
+    def __init__(self, required_role: Role, verify: bool = True):
+        self.required_role = required_role
+        self.verify = verify
+
+    def __call__(self, token=Depends(JWTBearer()), db=Depends(get_db)) -> bool:
+        role = get_user(db=db, username=get_username(token)).role
+        if role == Role.USER and self.required_role == Role.ADMIN:
+            if self.verify:
+                raise HTTPException(
+                    status_code=403,
+                    detail={
+                        "reason": "You need to be administrator for this transaction.",
+                    },
+                )
+            else:
+                return False
+        return True
+
+
+class ProjectRoleVerification:
+    roles = [
+        ProjectUserRole.USER,
+        ProjectUserRole.MANAGER,
+        ProjectUserRole.ADMIN,
+    ]
+
+    def __init__(
+        self,
+        required_role: ProjectUserRole,
+        verify: bool = True,
+        required_permission: ProjectUserPermission = None,
+    ):
+        self.required_role = required_role
+        self.verify = verify
+        self.required_permission = required_permission
+
+    def __call__(
+        self, project_slug: str, token=Depends(JWTBearer()), db=Depends(get_db)
+    ) -> bool:
+        user = get_user(db=db, username=get_username(token))
+
+        # TODO: Use slug directly
```

Comment:
`project_users.get_user_of_project` has to be refactored in the future - right now it still depends on the project name, but it should use the slug. This is a major refactoring because it involved the database as well - that's the reason why I don't want to handle it as part of this PR. 

Line `66` could be removed when we can pass the project slug to `get_project_by_slug` directly: 
```
project_user = project_users.get_user_of_project(
       db, project_slug=project_slug, user_id=user.id
)
```